lxd/devices.go

```diff
@@ -823,3 +979,71 @@ func deviceParseDiskLimit(readSpeed string, writeSpeed string) (int64, int64, in
 
 	return readBps, readIops, writeBps, writeIops, nil
 }
+
+const USB_PATH = "/sys/bus/usb/devices"
+
+func loadRawValues(p string) (map[string]string, error) {
+	values := map[string]string{
+		"idVendor":  "",
+		"idProduct": "",
+		"dev":       "",
+		"busnum":    "",
+		"devnum":    "",
+	}
+
+	for k, _ := range values {
+		v, err := ioutil.ReadFile(path.Join(p, k))
+		if err != nil {
+			return nil, err
+		}
+
+		values[k] = strings.TrimSpace(string(v))
+	}
+
+	return values, nil
+}
+
+func deviceLoadUsb() ([]usbDevice, error) {
+	result := []usbDevice{}
+
+	ents, err := ioutil.ReadDir(USB_PATH)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
On Wed, Aug 03, 2016 at 11:19:27AM -0700, Stéphane Graber wrote:
> > +			return nil, err
> > +		}
> > +
> > +		values[k] = strings.TrimSpace(string(v))
> > +	}
> > +
> > +	return values, nil
> > +}
> > +
> > +func deviceLoadUsb() ([]usbDevice, error) {
> > +	result := []usbDevice{}
> > +
> > +	ents, err := ioutil.ReadDir(USB_PATH)
> > +	if err != nil {
> > +		return nil, err
> > +	}
> 
> What happens on a system without USB? Does that only cause a failure when a usb device is added to the container or does that also affect containers without usb devices?

I *think* I've done it so that it would only fail on systems where
there usb devices configured on containers. that said, it's not hard
to add a little check in there for ENOENT and return an empty list, so
i'll do that.

> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/2261/files/7967263fb5386ef2d2c876f1bb02d84c9d5a579a#r73390271